skimage/measure/_colocalization.py

```diff
@@ -0,0 +1,371 @@
+
+
+def intersection_coefficient(imgA_mask, imgB_mask, roi=None):
+    """Fraction of a channel's segmented binary mask that overlaps with a
+    second channel's segmented binary mask.
+
+    Parameters
+    ----------
+    imgA_mask : (M, N) ndarray of dtype bool
+        Image of channel A.
+    imgB_mask : (M, N) ndarray of dtype bool
+        Image of channel B.
+        Must have same dimensions as `imgA_mask`.
+    roi : (M, N) ndarray of dtype bool, optional
+        Only `imgA_mask` and `imgB_mask` pixels within this region of interest
+        mask are included in the calculation.
+        Must have same dimensions as `imgA_mask`.
+
+    Returns
+    -------
+    Intersection coefficient, float
+        Fraction of `imag1_mask` that overlaps with `imgB_mask`.
+
+    """
+    if roi is None:
+        roi = np.ones_like(imgA_mask)
+    check_numpy_arr(imgA_mask, 'imgA_mask', bool_expected=True)
+    check_numpy_arr(imgB_mask, 'imgB_mask', bool_expected=True)
+    check_numpy_arr(roi, 'roi', bool_expected=True)
+    check_shape_equality_all(imgA_mask, imgB_mask, roi)
+
+    imgA_mask = imgA_mask[roi.astype(bool)]
+    imgB_mask = imgB_mask[roi.astype(bool)]
+    if np.count_nonzero(imgA_mask) == 0:
+        return 0
+    denom = np.count_nonzero(imgA_mask)
+    return np.count_nonzero(np.logical_and(imgA_mask, imgB_mask)) / denom
+
+
+def pixel_intensity_sum(img, roi=None):
+    """Sum of all intensity values of image within the ROI.
+
+    Parameters
+    ----------
+    img : (M, N) ndarray
+    roi : (M, N) ndarray of dtype bool, optional
+        Only `img` pixel values within this region of interest mask are
+        included in the calculation.
+        Must have same dimensions as `img`.
+
+    Returns
+    -------
+    float
+        Sum of all intensity values of image within the ROI.
+
+    Notes
+    -------
+    Equivalent to "Integrated Density" "in ImageJ [1]_.
+
+    References
+    ----------
+    .. [1] https://imagej.nih.gov/ij/docs/menus/analyze.html
+
+    """
+    if roi is None:
+        roi = np.ones_like(img)
+    check_numpy_arr(img, 'img', bool_expected=False)
+    check_numpy_arr(roi, 'roi', bool_expected=True)
+    check_shape_equality_all(img, roi)
+
+    img = img[roi.astype(bool)]
+    return np.sum(img)
+
+
+def av_pixel_intensity(img, roi=None):
+    """ Average intensity of all pixels within the ROI.
+
+    Parameters
+    ----------
+    img : (M, N) ndarray
+    roi : (M, N) ndarray of dtype bool, optional
+        Only `img` pixel values within this region of interest mask are
+        included in the calculation.
+        Must have same dimensions as `img`.
+
+    Returns
+    -------
+    float
+
+    """
+    if roi is None:
+        roi = np.ones_like(img)
+    check_numpy_arr(img, 'img', bool_expected=False)
+    check_numpy_arr(roi, 'roi', bool_expected=True)
+    check_shape_equality_all(img, roi)
+
+    img = img[roi.astype(bool)]
+    return np.sum(img) / np.size(img)
```

Comment:
I am not sure we should bother to provide dedicated functions for `av_pixel_intensity` or `pixel_intensity_sum`. That should be easy for users to get on their own. For example, this function is basically just `img[roi].mean()` with some shape checks.